Shape_detection/examples/Shape_detection/region_growing_on_polygon_mesh.cpp

```diff
@@ -1,92 +1,57 @@
     using Sorting        = CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_sorting<Kernel, Polygon_mesh, Neighbor_query>;
-
 #else
-
-    using Polygon_mesh = CGAL::Polyhedron_3<Kernel, CGAL::Polyhedron_items_3, CGAL::HalfedgeDS_vector>;
-    using Face_range   = typename CGAL::Iterator_range<typename boost::graph_traits<Polygon_mesh>::face_iterator>;
-
+    using Polygon_mesh   = CGAL::Polyhedron_3<Kernel, CGAL::Polyhedron_items_3, CGAL::HalfedgeDS_vector>;
+    using Face_range     = typename CGAL::Iterator_range<typename boost::graph_traits<Polygon_mesh>::face_iterator>;
     using Neighbor_query = CGAL::Shape_detection::Polygon_mesh::One_ring_neighbor_query<Polygon_mesh, Face_range>;
     using Region_type    = CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_region<Kernel, Polygon_mesh, Face_range>;
     using Sorting        = CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_sorting<Kernel, Polygon_mesh, Neighbor_query, Face_range>;
-
 #endif
 
-using Region  = std::vector<std::size_t>;
-using Regions = std::vector<Region>;
-
 using Vertex_to_point_map = typename Region_type::Vertex_to_point_map;
-
-using Region_growing = CGAL::Shape_detection::Region_growing<Face_range, Neighbor_query, Region_type, typename Sorting::Seed_map>;
+using Region_growing      = CGAL::Shape_detection::Region_growing<Face_range, Neighbor_query, Region_type, typename Sorting::Seed_map>;
 
 int main(int argc, char *argv[]) {
 
-  std::cout << std::endl <<
-    "region_growing_on_polygon_mesh example started"
-  << std::endl << std::endl;
-
-  // Load off data either from a local folder or a user-provided file.
-  std::ifstream in(argc > 1 ? argv[1] : "data/polygon_mesh.off");
-  CGAL::set_ascii_mode(in);
+  // Load data either from a local folder or a user-provided file.
+  const std::string input_filename = (argc > 1 ? argv[1] : "data/polygon_mesh.off");
+  std::ifstream in_off(input_filename);
+  std::ifstream in_ply(input_filename);
+  CGAL::set_ascii_mode(in_off);
+  CGAL::set_ascii_mode(in_ply);
 
-  if (!in) {
-    std::cout <<
-    "Error: cannot read the file polygon_mesh.off!" << std::endl;
-    std::cout <<
-    "You can either create a symlink to the data folder or provide this file by hand."
-    << std::endl << std::endl;
+  Polygon_mesh polygon_mesh;
+  if (CGAL::read_OFF(in_off, polygon_mesh)) { in_off.close();
+  } else if (CGAL::read_PLY(in_ply, polygon_mesh)) { in_ply.close();
+  } else {
+    std::cerr << "ERROR: cannot read the input file!" << std::endl;
     return EXIT_FAILURE;
   }
-
-  Polygon_mesh polygon_mesh;
-  in >> polygon_mesh;
-
-  in.close();
   const Face_range face_range = faces(polygon_mesh);
-
-  std::cout <<
-    "* polygon mesh with "
-  << face_range.size() <<
-    " faces is loaded"
-  << std::endl;
+  std::cout << "* number of input faces: " << face_range.size() << std::endl;
+  assert(face_range.size() == 32245);
```

Comment:
unlikely to work with any input

Reply:
True. I was testing on my data. I plan to remove asserts from examples.